test/PoolManager.ts

```diff
@@ -0,0 +1,44 @@
+import { expect } from "chai";
+import { buildContractTestContext, ContractTestContext } from "./ContractTestContext";
+import { parseUnits, formatUnits } from "ethers/lib/utils";
+
+const oneK18Decimal = parseUnits("1000", 18);
+const twentyK18Decimal = parseUnits("20000", 18);
+
+function getFloatFromBigNum (bigNumValue, dec = 18) {
+    return parseFloat(formatUnits(bigNumValue, dec));
+}
+
+describe("PoolManager test suit", async function () {
+    let r: ContractTestContext;
+
+    before(async () => {
+        r = await buildContractTestContext();
+
+        await r.lvUSD.setMintDestination(r.coordinator.address);
+        // mint a bit lvUSD
+        await r.lvUSD.mint(twentyK18Decimal);
+
+        // Approve pool manager to access 3CRV and lvUSD funds
+        await r.external3CRV.approve(r.poolManager.address, oneK18Decimal);
+        await r.lvUSD.approve(r.poolManager.address, oneK18Decimal);
+    });
+
+    it("Should have enough funds in account and coordinator", async function () {
+        /// Sanity check for funds in account. Does not test anything in Pool Manager
+        expect(getFloatFromBigNum(await r.external3CRV.balanceOf(r.owner.address))).to.be.gt(3400);
+        expect(getFloatFromBigNum(await r.lvUSD.balanceOf(r.coordinator.address))).to.be.eq(20000);
+    });
+
+    it("Should fail to fund pool if not enough lvUSD on coordinator", async function () {
+        const tooMuchLvUSDAmount = parseUnits("200000");
+        const promise = r.poolManager.fundPoolWith3CRV(r.owner.address, tooMuchLvUSDAmount);
+        await expect(promise).to.be.revertedWith("Insufficient lvUSD on Coord");
+    });
+
+    it("Should fund pool with both lvUSD and USDC", async function () {
+        await r.poolManager.fundPoolWith3CRV(r.owner.address, oneK18Decimal);
+        expect(getFloatFromBigNum(await r.curveLvUSDPool.balances(0))).to.be.eq(1200);
```

Comment:
I am not sure why `eq 1200`?

I would suggest getting the balance before and after and then calc the diff. The mainnet fork block might change in the future and might fail the test. Also we might forget the context here and will have hard time figuring what we have done

Reply:
yep - but the pool isn't necessarily going to be empty in the future

Reply:
I understand the want to make the tests more general but its a balance between readability and making sure "wrong" numbers fails the test. However, I think there is room for both as the non general test approach (ie hard numbers) gives us some intuition on code and state. I think a healthy mix if both is best 

Its the pool we spin up with textContext so we know how much it has in each test